src/main/java/com/gregtechceu/gtceu/api/machine/trait/NotifiableFluidTank.java

```diff
@@ -118,31 +118,24 @@ public List<FluidIngredient> handleRecipeInner(IO io, GTRecipe recipe, List<Flui
             }
 
             FluidStack[] fluids;
-            int amount;
 
             if (io == IO.OUT && ingredient instanceof IntProviderFluidIngredient provider) {
                 provider.setFluidStacks(null);
                 provider.setSampledCount(-1);
 
                 if (simulate) {
                     fluids = new FluidStack[] { provider.getMaxSizeStack() };
-                    amount = provider.getCountProvider().getMaxValue();
                 } else {
                     fluids = provider.getStacks();
-                    if (fluids.length == 0 || fluids[0].isEmpty()) {
-                        it.remove();
-                        continue;
-                    }
-                    amount = fluids[0].getAmount();
                 }
             } else {
                 fluids = ingredient.getStacks();
-                if (fluids.length == 0 || fluids[0].isEmpty()) {
-                    it.remove();
-                    continue;
-                }
-                amount = ingredient.getAmount();
```

Comment:
When simplifying this action, this case got lost and now instead becomes fluids[0].getAmount() instead of ingredient.getAmount(). Was that on purpose?